Update Dockerfile and docker-compose

sorbaugh opened the pull request:
* Resolves: #37169

## Summary
Proposed change to in-built docker-compose.yml and Dockerfile. The current implementation uses php 7.4, which is no longer compatible with the required PHP version of the server. I upped this to PHP 8.1

List of changes:
- Upped PHP Version to 8.1
- QoL: Added Apache Webserver so the Container works "out of the box" after docker-compose up -d
- Mounting whole project as volume to /var/www/html in docker-compose.yml (and set WORKDIR to /var/www/html)

Tested in a Docker for Windows environment.

How to run: 
- Navigate to .devcontainer directory in terminal and execute 
`docker-compose up -d`
- After container is started. Restart Apache Service in Container
`docker exec devcontainer-nextclouddev-1 service apache2 restart`
- Initialize Submodules (Only necessary for first time setup)
`docker exec devcontainer-nextclouddev-1 git submodule update --init`

After Container starts, open in Browser:
`http://localhost/`

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

Commit:
Proposed change to in-build docker-compose.yml and Dockerfile. The current implementation uses php 7.4, which is no longer compatible with the required PHP version of the server. I upped this to PHP 8.1

List of changes:
- Upped PHP Version to 8.1
- Added Apache Webserver so the Container works "out of the box" after docker-compose up -d
- Mounting whole project as volume to /var/www/html in docker-compose.yml (and set WORKDIR to /var/www/html)

Tested in a Docker for Windows environment.

R0Wi committed:
Minor improvements for devcontainer

* Autostart apache2
* Apply occ installation on start
* Autostart Xdebug on request
* Add DevContainer Xdebug profile

R0Wi commented:
LGTM so far. I added some minor improvements in 6bfa5d42485820cd5712ecf23f235f4d61f5031c . It should now be possible to just open a fresh Codespaces instance and start debugging right away. Maybe you want to pick this commit to your branch as well?

skjnldsv commented on .devcontainer/Dockerfile:
```diff
@@ -4,31 +4,43 @@ ARG DEBIAN_FRONTEND=noninteractive
 # PHP
 RUN apt-get update -y
+
+RUN apt install -y apache2 vim software-properties-common
+
+RUN add-apt-repository ppa:ondrej/php
```
this is using focal
![image](https://user-images.githubusercontent.com/14975046/224935800-6333a3a5-ec43-4008-9cc3-592a53f89425.png)

szaimen replied:
Ah I meant jammy

R0Wi replied:
Resolved by e6996140bf28ccb8e08f31f27dc5a6ee860036d1

artonge commented on .devcontainer/Dockerfile:
```diff
@@ -44,4 +56,6 @@ RUN add-apt-repository \
 RUN apt-get update -y
 RUN apt-get install -y docker-ce docker-ce-cli containerd.io
-RUN ln -s /var/run/docker-host.sock /var/run/docker.sock  
+RUN ln -s /var/run/docker-host.sock /var/run/docker.sock
```
Any reason this is needed ?
Not even sure why we are binding the docker socket in the first place.

R0Wi committed:
Use ubuntu jammy and cleanup Dockerfile

R0Wi commented:
I added the following:

-  211c5ec398a89413a943b432051b89585ed349bc : I'm now able to directly open and debug the NC project after codespaces start without any futher steps
- e6996140bf28ccb8e08f31f27dc5a6ee860036d1 : Use the official jammy base image to be able to use PHP 8.1 without 3rd party repos

sorbaugh committed:
Merge branch 'nextcloud:master' into master

Commit:
Fixes Issue 35286 Firefox inconsistent scrolling

URL: https://github.com/nextcloud/server/issues/35286

Adds overflow-anchor: none to #app-content element within files.scss and re-minified files.css and merged.css.

Tested on Firefox 111.0 (64-Bit)

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	apps/files/css/files.css
#	apps/files/css/merged.css

Commit:
Fixes Issue 35286 Firefox inconsistent scrolling

URL: https://github.com/nextcloud/server/issues/35286

Adds overflow-anchor: none to #app-content element within files.scss and re-minified files.css and merged.css.

Tested on Firefox 111.0 (64-Bit)

R0Wi commented:
@sorbaugh any reason why this has been closed?
